File: src/PHP/BehavioralPatterns/Mediator/CommunicationHub/Participants/Sensor.php
<?php

namespace BehavioralPatterns\Mediator\CommunicationHub\Participants;

class Sensor extends Participant
{
    private string $id;

    protected float $lastValue;

    public function __construct($id)
    {
        $this->id = $id;
    }

    public function getId(): string
    {
        return $this->id;
    }

    public function getLastValue(): float
    {
        return $this->lastValue;
    }

    public function receive($sender, $args): void
    {
        if ($args === "measure")
        {
            $this->lastValue = mt_rand() / mt_getrandmax();
            $this->mediator->notify($this, $this->lastValue);
        }
    }

    public function valueChanged($value): void
    {
        $this->lastValue = $value;
        $this->mediator->notify($this, $this->lastValue);
    }

    public function __toString()
    {
        return "Sensor({$this->id})";
    }
}

File: src/C#/BehavioralPatterns/Mediator/MultipleExamples/CommunicationHub/Participants/Sensor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MultipleExamples.CommunicationHub.Participants
{
    public class Sensor : Participant
    {
        public Sensor(string id)
        {
            Id = id;
        }

        public string Id { get; private set; }

        public object LastValue { get; protected set; }

        public override void Receive(Participant sender, object args)
        {
            if (args == "measure")
            {
                LastValue = new Random().NextDouble();
                Mediator.Notify(this, LastValue);
            }
        }

        public virtual void ValueChanged(object value)
        {
            LastValue = value;
            Mediator.Notify(this, LastValue);
        }

        public override string ToString() => $"Sensor({Id})";
    }
}
